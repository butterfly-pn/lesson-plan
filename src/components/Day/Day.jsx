import React from "react";
import Subject from "../Subject/Subject";

import styles from "./style.module.scss";

/**
 * @name Day
 * @description Lesson plan for day
 * @example
 * {weekDays.map(day => (
 * <Day
 *   key={day}
 *   day={day}
 *   subjects={this.state.plan.map(row => {
 *     // Filter row to display only data from current day.
 *     const rowData = row[weekDaysToPolish[day]];
 *     let subject = rowData.split(" ");
 *     const classroom = subject.pop();
 *     return {
 *       key: row.Godz + rowData,
 *       name: rowData.slice(0, rowData.length - classroom.length),
 *       classroom: classroom,
 *       hour: row.Godz.split("-")[0]
 *     };
 *   })}
 * />
 *))}
 */
export default class Day extends React.Component {
  render() {
    return (
      <div className={styles.day} id={this.props.id}>
        <h2>{this.props.day}</h2>
        {this.props.subjects.map((subject, i) => (
          <>
            {subject.name ? (
              <Subject
                key={subject.key}
                name={subject.name}
                classroom={subject.classroom}
                hour={subject.hour}
                color="#fa77fc"
              />
            ) : (
              <div className={styles.free} />
            )}
          </>
        ))}
      </div>
    );
  }
}
