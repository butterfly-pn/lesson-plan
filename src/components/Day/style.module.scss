.day {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.free {
  height: 124px;
  margin: 8px;
  box-sizing: border-box;
  display: none;
}

.free:first-of-type {
  display: none;
}

@media (min-width: 1250px) {
  .free {
    display: block;
  }
}
