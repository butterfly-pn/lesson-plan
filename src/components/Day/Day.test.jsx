import React from "react";
import ReactDOM from "react-dom";
import Day from "./Day";
import renderer from "react-test-renderer";

describe("<Day />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Day subjects={[{ name: "Math", classroom: "16" }]} />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Snapshot", () => {
    const component = renderer.create(
      <Day subjects={[{ name: "Math", classroom: "16" }]} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
