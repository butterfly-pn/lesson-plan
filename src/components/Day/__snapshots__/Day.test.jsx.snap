// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Day /> Snapshot 1`] = `
<div
  className="day"
>
  <h2 />
  <div
    className="sc-bdVaJa lcmtfW"
  >
    <p
      className="subject-name"
    >
      Math
    </p>
    <p
      className="subject-classroom"
    >
      16
    </p>
    <div
      className="hour_wrapper"
    >
      <div
        className="hour"
      />
    </div>
  </div>
</div>
`;
